### PR TITLE
Lesgo: fix lesgo types definition bugs for 0.6.0

### PR DESCRIPTION
**Pull request by xXLXx**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [https://reflex-media.github.io/lesgo-docs/](https://reflex-media.github.io/lesgo-docs/)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/lesgo/utils/validateFields.d.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 export interface Validation<T> {
-    key: T;
-    type: "string" | "object" | "number" | "decimal" | "email" | "array" | "enum";
-    required?: boolean;
+	key: T;
```
Comment (DangerBotOSS) [inline types/lesgo/utils/validateFields.d.ts:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by DangerBotOSS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/lesgo/utils/prepSQLInsertParams.d.ts#L4** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/lesgo/utils/prepSQLUpdateParams.d.ts#L4** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  </tbody>
</table>



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## lesgo ([<kbd>unpkg</kbd>](https://unpkg.com/browse/lesgo@latest/))
was missing the following properties:
1. httpMiddleware
1. normalizeHttpRequestMiddleware
1. successHttpResponseMiddleware
1. errorHttpResponseMiddleware
1. normalizeSQSMessageMiddleware

<details>
<summary>as well as these 12 other properties...</summary>
<p>verifyJwtMiddleware, AuroraDbService, DynamoDbService, ElastiCacheService, ElasticsearchService, JwtService, LoggerService, S3Service, SQSService, cache, elasticsearch, crypto</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cb9bd99ddf0faaefbbc0267a8767d5e74fa6ece9
</p>

File: types/lesgo/tests/services/S3Service.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
-import S3Service from "lesgo/services/S3Service";
+import S3Service from 'lesgo/services/S3Service';
 
 // $ExpectType S3Service
 const s3 = new S3Service({
-    accessKeyId: "AKID",
-    secretAccessKey: "SECRET",
-    region: "us-west-2",
+	accessKeyId: 'AKID',
```
Comment (DangerBotOSS) [inline types/lesgo/tests/services/S3Service.ts:5]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by 3773nortiz — APPROVED** [review]

**Comment by xXLXx** [comment]
Ready to merge

**Pull request merged**